atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraTimestampsEteTest.java

```diff
@@ -87,12 +87,42 @@ public void timestampsForRangeTombstonesAreCurrentInThoroughTables() throws IOEx
         assertDroppableTombstoneRatioPositive(sstableMetadata);
     }
 
-    private void sweepUntilNoValueExistsAtTimestamp(long id, long timestamp) {
+    @Test
+    public void timestampsForSentinelsAndTombstonesAreCurrentInConservativeTables()
+            throws IOException, InterruptedException {
+        long firstWriteTimestamp = todoClient.addNamespacedTodoWithIdAndReturnTimestamp(ID, NAMESPACE, TODO);
+        todoClient.addNamespacedTodoWithIdAndReturnTimestamp(ID, NAMESPACE, TODO_2);
+        sweepUntilNoValueExistsAtTimestamp(ID, firstWriteTimestamp, Optional.of(NAMESPACE));
+
+        CassandraCommands.nodetoolFlush(CASSANDRA_CONTAINER_NAME);
+
+        List<String> ssTables = CassandraCommands.nodetoolGetSSTables(
+                CASSANDRA_CONTAINER_NAME,
+                "atlasete",
+                TodoSchema.namespacedTodoTable(),
+                ValueType.STRING.convertFromJava(NAMESPACE));
+        String sstableMetadata = CassandraCommands.ssTableMetadata(
+                CASSANDRA_CONTAINER_NAME, Iterables.getOnlyElement(ssTables));
+
+        // The table should contain the first value, and timestamps thereafter should be higher.
+        // Failure here means either that the range tombstone was written at a lower Cassandra timestamp than the
+        // original value (unlikely since we confirmed no value existed at the first write timestamp), or
+        // that the sweep sentinel was written at a lower Cassandra timestamp than the original value (and hence not
+        // a fresh timestamp).
+        assertMinimumTimestampIsAtLeast(sstableMetadata, firstWriteTimestamp);
+
+        // Failure here means that the range tombstone isn't droppable, which would be weird
```

Comment:
This would happen if the range tombstone was written at the wrong C* timestamp, right?

Reply:
If I'm not wrong right now these are just calculated as a ratio of tombstones to values - see https://issues.apache.org/jira/browse/CASSANDRA-13024. So this would be 0 if the tombstone was somehow already dropped which shouldn't happen. I've clarified the comment - `weird` isn't a good description.